app/src/main/java/org/mozilla/focus/widget/DownloadListAdapter.java

```diff
@@ -158,7 +158,12 @@ private void addItem(DownloadInfo downloadInfo){
 
         if (index == -1){
             mDownloadInfo.add(downloadInfo);
-            this.notifyItemInserted(mDownloadInfo.size()-1);
+            if (mDownloadInfo.size() > 1){
```

Comment:
Are we giving the incorrect index?
Simply fix with
`this.notifyItemInserted(mDownloadInfo.size());` ?